ProcessLib/Output.cpp

```diff
@@ -249,10 +235,32 @@ void Output::doOutputNonlinearIteration(Process const& process,
         return;
     }
 
+    const bool make_output =
+        !(process_id < static_cast<int>(_single_process_data.size()) - 1 &&
+          !(process.isMonolithicSchemeUsed()));
+    if (!make_output)
+        return;
+
     BaseLib::RunTime time_output;
     time_output.start();
 
-    findSingleProcessData(process, process_id);
+    auto spd_range = _single_process_data.equal_range(&process);
```

Comment:
Maybe this entire green block could be moved to a separate method `SPD* getSingleProcessData(int process_id)` and also used above.

Reply:
Added a new member function for that.  Since the new member can not have const modifier,  the const modifer of `doOutputNonlinearIteration` was dropped.